crates/recorder/src/measure/counters.rs

```diff
@@ -73,15 +75,17 @@ impl Measure for CounterMeasure {
         self.event_group.enable().unwrap()
     }
 
-    fn end(&mut self) -> Measurement {
-        self.event_group.disable().unwrap();
+    fn end(&mut self, phase: Phase, measurements: &mut Measurements) {
         let counts = self.event_group.read().unwrap();
-        Measurement::PerfCounters(PerfCounters {
-            cpu_cycles: counts[&self.cpu_cycles],
-            instructions_retired: counts[&self.instructions_retired],
-            cache_accesses: counts[&self.cache_accesses],
-            cache_misses: counts[&self.cache_misses],
-        })
+        measurements.reserve(4);
+        measurements.add(phase, "cpu-cycles".into(), counts[&self.cpu_cycles]);
+        measurements.add(
+            phase,
+            "instructions-retired".into(),
+            counts[&self.instructions_retired],
+        );
+        measurements.add(phase, "cache-accesses".into(), counts[&self.cache_accesses]);
+        measurements.add(phase, "cache-misses".into(), counts[&self.cache_misses]);
```

Comment:
This type of thing could be moved to a `Measurement::events(&self) -> Vec<(String, u64) ` function.